core/lib/zksync_core/src/state_keeper/batch_executor/tests/mod.rs

```diff
@@ -29,15 +38,86 @@ fn assert_reverted(execution_result: &TxExecutionResult) {
     }
 }
 
-/// Checks that we can successfully execute a single L2 tx in batch executor.
+#[derive(Debug, Clone)]
+enum StorageType {
+    AsyncRocksdbCache,
+    Rocksdb,
+    Postgres,
+}
+
+impl StorageType {
+    const ALL: [Self; 3] = [Self::AsyncRocksdbCache, Self::Rocksdb, Self::Postgres];
+
+    async fn create_batch_executor(&self, tester: &Tester) -> BatchExecutorHandle {
+        let (l1_batch_env, system_env) = tester.default_batch_params();
+        match self {
+            StorageType::AsyncRocksdbCache => tester.create_batch_executor().await,
```

Comment:
Bikeshedding: the methods in `StorageType` look like they belong to `Tester`, since they mostly operate on it, if this makes sense.